components/brave_rewards/resources/tip/components/monthly_tip_form.tsx

```diff
@@ -0,0 +1,166 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+
+import { PaymentKind, RewardsParameters, PublisherInfo } from '../lib/interfaces'
+import { HostContext } from '../lib/host_context'
+import { LocaleContext } from '../lib/locale_context'
+import { formatExchangeAmount } from '../lib/formatting'
+
+import { CurrentMonthlyForm } from './current_monthly_form'
+import { FormSubmitButton } from './form_submit_button'
+import { PaymentKindSwitch } from './payment_kind_switch'
+import { TipAmountSelector } from './tip_amount_selector'
+import { BatString } from './bat_string'
+import { TermsOfService } from './terms_of_service'
+import { CalendarIcon } from './icons/calendar_icon'
+
+import * as style from './monthly_tip_form.style'
+
+function generateTipOptions (
+  rewardsParameters: RewardsParameters,
+  publisherInfo: PublisherInfo
+) {
+  const publisherAmounts = publisherInfo.amounts
+  if (publisherAmounts && publisherAmounts.length > 0) {
+    return publisherAmounts
+  }
+  const { monthlyTipChoices } = rewardsParameters
+  if (monthlyTipChoices.length > 0) {
+    return monthlyTipChoices
+  }
+  return [1, 5, 10]
+}
```

Comment:
This looks pretty similar to the one-time tip function, could we share the implementation between the two components?